hs_core/models.py

```diff
@@ -1898,6 +1900,35 @@ def delete_all_elements(self):
         self.relations.all().delete()
         self.funding_agencies.all().delete()
 
+    def db_instance_to_dict(self, instance, fields=None, exclude=None):
+        """
+        modified version of Django's model_to_dict method accomodates many_to_many relations
+
+        Returns a dict containing the data in ``instance`` suitable for passing as
+        a Form's ``initial`` keyword argument.
+
+        ``fields`` is an optional list of field names. If provided, only the named
+        fields will be included in the returned dict.
+
+        ``exclude`` is an optional list of field names. If provided, the named
+        fields will be excluded from the returned dict, even if they are listed in
+        the ``fields`` argument.
+        """
+        opts = instance._meta
+        data = {}
+        for f in chain(opts.concrete_fields, opts.many_to_many):
```

Comment:
@pkdash you are correct. I did what you suggested and it worked. here is the modified _add_swat_objective method:

 @classmethod
    def _add_swat_objective(cls, model_objective, objectives):
        for swat_objective in objectives:
            if isinstance(swat_objective, int):
                qs = ModelObjectiveChoices.objects.filter(id=swat_objective)
            else:
                qs = ModelObjectiveChoices.objects.filter(description__iexact=swat_objective)
            if qs.exists():
                model_objective.swat_model_objectives.add(qs[0])
            else:
                model_objective.swat_model_objectives.create(description=swat_objective)

If you like it I will do a PR with the fix.